observer/probers/tls/tls_conf.go

```diff
@@ -0,0 +1,149 @@
+package probers
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	certExpiryName = "obs_cert_expiry"
+	statusCodeName = "tls_prober_status_code"
+)
+
+// TLSConf is exported to receive YAML configuration.
+type TLSConf struct {
+	URL      string `yaml:"url"`
+	Root     string `yaml:"root"`
+	Response string `yaml:"response"`
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Configurer`.
+func (c TLSConf) Kind() string {
+	return "TLS"
+}
+
+// UnmarshalSettings takes YAML as bytes and unmarshals it to the to an
+// TLSConf object.
+func (c TLSConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf TLSConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c TLSConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, expected a valid url", c.URL)
+	}
+	if url.Scheme == "" {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, missing scheme", c.URL)
+	}
+	return nil
+}
+
+func (c TLSConf) validateRoot() error {
+	// expected example: /O=Internet Security Research Group/CN=ISRG Root X1
+	regex, err := regexp.Compile("^/O=[^/]*/CN=[^/]*$")
+	if err != nil {
+		return err
+	}
+	matched := regex.MatchString(c.Root)
+	if matched {
+		return nil
+	} else {
+		return fmt.Errorf(
+			"invalid 'root`, got: %s\nDid not match expected format: '/CN:*/O:*`", c.Root)
+	}
+}
+
+func (c TLSConf) validateResponse() error {
+	acceptable := []string{"valid", "expired", "revoked"}
+	for _, a := range acceptable {
+		if strings.ToLower(c.Response) == a {
+			return nil
+		}
+	}
+	return fmt.Errorf(
+		"invalid `response`, got: %s\nMust be one of 'valid', 'expired' or 'revoked'.", c.Response)
```

Comment:
Go errors shouldn't be capitalized and shouldn't contain newlines. Also we should just print the string representation of `acceptable` so that we don't have to update the error and the list separately. 

```suggestion
		"invalid response, got %q . Must be one of %s", c.Response, acceptable)
```